spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -112,25 +96,24 @@ public SleuthProperties stackdriverSleuthProperties(SleuthProperties sleuthPrope
 
 	@Bean
 	@ConditionalOnMissingBean
-	public Reporter<zipkin2.Span> reporter(
-			FlushableTraceConsumer traceConsumer,
-			SpanTranslator spanTranslator) {
-		return new StackdriverTraceReporter(
-				this.finalProjectIdProvider.getProjectId(),
-				traceConsumer,
-				spanTranslator);
+	public Reporter<Span> reporter(Sender sender, BytesEncoder<Span> encoder) {
+		return AsyncReporter.builder(sender).build(encoder);
 	}
 
 	@Bean
 	@ConditionalOnMissingBean
-	public SpanTranslator spanTranslator(LabelExtractor labelExtractor) {
-		return new SpanTranslator(labelExtractor);
+	public Sender stackdriverSender() throws IOException {
+		return StackdriverSender.newBuilder()
+				.projectId(this.finalProjectIdProvider.getProjectId())
+				.callOptions(CallOptions.DEFAULT.withCallCredentials(
```

Comment:
I think we need to expose the rest of the `CallOptions` through `GcpTraceProperties`.

Reply:
Shall we do that in another PR?
There are a lot of parameters to be exposed.

Reply:
Sure, feel free to create a linked issue.

Reply:
https://github.com/spring-cloud/spring-cloud-gcp/issues/662